modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/AbstractDeployableArtifact.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.impl.internal.artifact;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.capitalize;
+import static org.mule.runtime.core.api.util.ClassUtils.withContextClassLoader;
+import static org.mule.runtime.core.internal.logging.LogUtil.log;
+import static org.mule.runtime.core.internal.util.splash.SplashScreen.miniSplash;
+import org.mule.runtime.api.lifecycle.Stoppable;
+import org.mule.runtime.deployment.model.api.DeployableArtifact;
+import org.mule.runtime.deployment.model.api.DeployableArtifactDescriptor;
+import org.mule.runtime.deployment.model.api.DeploymentStopException;
+import org.mule.runtime.deployment.model.api.artifact.ArtifactContext;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.DisposableClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.RegionClassLoader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Common behaviour for deployable artifacts to handle the stop/dispose phases.
+ */
+public abstract class AbstractDeployableArtifact<D extends DeployableArtifactDescriptor> implements DeployableArtifact<D> {
+
+  protected transient final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the "transient" modifier from this field. ([squid:S2065](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2065))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7686&issues=AWjzYs1-otIXnmsUaOza&open=AWjzYs1-otIXnmsUaOza)

[ISSUE_KEY]: AWjzYs1-otIXnmsUaOza